Unused parameter warnings

```
In file included from ../src/systems/system_projection.C:29:
In file included from ../contrib/metaphysicl/0.2.0/src/numerics/include/metaphysicl/dynamicsparsenumberarray_decl.h:32:
In file included from ../contrib/metaphysicl/0.2.0/src/numerics/include/metaphysicl/dynamicsparsenumberbase_decl.h:39:
../contrib/metaphysicl/0.2.0/src/core/include/metaphysicl/ct_set.h:721:44: warning: unused parameter 'i' [-Wunused-parameter]
  static int runtime_value_of(unsigned int i) {
                                           ^
In file included from ../src/systems/system_projection.C:29:
In file included from ../contrib/metaphysicl/0.2.0/src/numerics/include/metaphysicl/dynamicsparsenumberarray_decl.h:32:
In file included from ../contrib/metaphysicl/0.2.0/src/numerics/include/metaphysicl/dynamicsparsenumberbase_decl.h:43:
../contrib/metaphysicl/0.2.0/src/utilities/include/metaphysicl/testable.h:111:70: warning: unused parameter 'rhs' [-Wunused-parameter]
  inline void operator==(const safe_bool<T>& lhs,const safe_bool<U>& rhs) {
                                                                     ^
../contrib/metaphysicl/0.2.0/src/utilities/include/metaphysicl/testable.h:116:70: warning: unused parameter 'rhs' [-Wunused-parameter]
  inline void operator!=(const safe_bool<T>& lhs,const safe_bool<U>& rhs) {
                                                                     ^
  CXX      src/systems/libmesh_dbg_la-system_subset_by_subdomain.lo
In file included from ../src/systems/system_projection.C:80:
../contrib/metaphysicl/0.2.0/src/numerics/include/metaphysicl/dynamicsparsenumberarray.h:75:76: warning: unused parameter 'a' [-Wunused-parameter]
DynamicSparseNumberArray<T,I>::dot (const DynamicSparseNumberArray<T2,I2>& a) const
                                                                           ^
../contrib/metaphysicl/0.2.0/src/numerics/include/metaphysicl/dynamicsparsenumberarray.h:94:86: warning: unused parameter 'a' [-Wunused-parameter]
DynamicSparseNumberArray<T,I>::outerproduct (const DynamicSparseNumberArray<T2, I2>& a) const
                                                                                     ^
```

--- a/src/core/include/metaphysicl/ct_set.h
+++ b/src/core/include/metaphysicl/ct_set.h
@@ -718,7 +718,7 @@ struct NullContainer
   }
 
   template <typename T>
-  static int runtime_value_of(unsigned int i) {
+  static int runtime_value_of(unsigned int /*i*/) {
     throw std::domain_error("Container::runtime_value_of argument not found");
     return static_cast<unsigned int>(-1);
   }

--- a/src/numerics/include/metaphysicl/dynamicsparsenumberarray.h
+++ b/src/numerics/include/metaphysicl/dynamicsparsenumberarray.h
@@ -41,6 +41,8 @@ DynamicSparseNumberArray<T,I>::DynamicSparseNumberArray() {}
 template <typename T, typename I>
 inline
 DynamicSparseNumberArray<T,I>::DynamicSparseNumberArray(const T& val) {
+  // Avoid unused variable warnings in opt mode.
+  (void)val;
   // This makes no sense unless val is 0!
 #ifndef NDEBUG
   if (val)
@@ -52,6 +54,8 @@ template <typename T, typename I>
 template <typename T2>
 inline
 DynamicSparseNumberArray<T,I>::DynamicSparseNumberArray(const T2& val) {
+  // Avoid unused variable warnings in opt mode.
+  (void)val;
   // This makes no sense unless val is 0!
 #ifndef NDEBUG
   if (val)
@@ -72,7 +76,7 @@ inline
 DynamicSparseNumberArray
   <typename DotType<T,T2>::supertype,
    typename CompareTypes<I, I2>::supertype>
-DynamicSparseNumberArray<T,I>::dot (const DynamicSparseNumberArray<T2,I2>& a) const
+DynamicSparseNumberArray<T,I>::dot (const DynamicSparseNumberArray<T2,I2>& /*a*/) const
 {
   typedef typename DotType<T,T2>::supertype TS;
   typedef typename CompareTypes<I, I2>::supertype IS;
@@ -91,7 +95,7 @@ inline
 DynamicSparseNumberArray<
   typename OuterProductType<T,T2>::supertype,
   typename CompareTypes<I, I2>::supertype>
-DynamicSparseNumberArray<T,I>::outerproduct (const DynamicSparseNumberArray<T2, I2>& a) const
+DynamicSparseNumberArray<T,I>::outerproduct (const DynamicSparseNumberArray<T2, I2>& /*a*/) const
 {
   typedef typename OuterProductType<T,T2>::supertype TS;
   typedef typename CompareTypes<I, I2>::supertype IS;

--- a/src/numerics/include/metaphysicl/dynamicsparsenumberbase.h
+++ b/src/numerics/include/metaphysicl/dynamicsparsenumberbase.h
@@ -379,9 +379,8 @@ DynamicSparseNumberBase<T,I,SubType>::sparsity_intersection (const std::vector<I
   metaphysicl_assert(std::is_sorted(new_indices.begin(), new_indices.end()));
 #endif
 
-  typedef typename CompareTypes<I,I2>::supertype max_index_type;
-
 #ifndef NDEBUG
+  typedef typename CompareTypes<I,I2>::supertype max_index_type;
   typename std::vector<I>::iterator index_it = _indices.begin();
   typename std::vector<I2>::const_iterator index2_it = new_indices.begin();
 

--- a/src/utilities/include/metaphysicl/testable.h
+++ b/src/utilities/include/metaphysicl/testable.h
@@ -108,12 +108,12 @@ template <typename T>
  
  
 template <typename T, typename U> 
-  inline void operator==(const safe_bool<T>& lhs,const safe_bool<U>& rhs) {
+  inline void operator==(const safe_bool<T>& lhs,const safe_bool<U>& /*rhs*/) {
       lhs.this_type_does_not_support_comparisons();  
   }
  
 template <typename T,typename U> 
-  inline void operator!=(const safe_bool<T>& lhs,const safe_bool<U>& rhs) {
+  inline void operator!=(const safe_bool<T>& lhs,const safe_bool<U>& /*rhs*/) {
     lhs.this_type_does_not_support_comparisons();
   }
 

--- a/test/navier_unit.h
+++ b/test/navier_unit.h
@@ -19,17 +19,16 @@ static double evaluate_q (const Vector& xyz, const int);
 
 int main(void)
 {
-  int err = 0;
   int N   = 2; // mesh pts. in x and y
-  double su,sv,s2u,s2v,sp,se,s2e,s2p;
-  double pnorm, unorm, vnorm, enorm;
-  double pnorm_max, unorm_max, vnorm_max, enorm_max;
-  double prnorm_max = 0., urnorm_max = 0., vrnorm_max = 0., ernorm_max = 0.;
+  double s2u,s2v,s2e,s2p;
 
-  unorm_max = 0;
-  vnorm_max = 0;
-  pnorm_max = 0;
-  enorm_max = 0;
+#ifdef METAPHYSICL_HAVE_MASA
+  int err = 0;
+  double su,sv,sp,se;
+  double pnorm, unorm, vnorm, enorm;
+  double pnorm_max = 0., unorm_max = 0., vnorm_max = 0., enorm_max = 0.;
+  double prnorm_max = 0., urnorm_max = 0., vrnorm_max = 0., ernorm_max = 0.;
+#endif
 
   RawVector xvecinit(0), yvecinit(0);
   xvecinit.insert<0>() = 1.;

--- a/test/pde_unit.h
+++ b/test/pde_unit.h
@@ -21,17 +21,16 @@ using namespace MetaPhysicL;
 
 int main(void)
 {
-  int err = 0;
-  int N   = 10; // mesh pts. in x and y
-  double su,sv,s2u,s2v,sp,se,s2e,s2p;
-  double pnorm, unorm, vnorm, enorm;
-  double pnorm_max, unorm_max, vnorm_max, enorm_max;
-  double prnorm_max = 0., urnorm_max = 0., vrnorm_max = 0., ernorm_max = 0.;
+  int N = 10; // mesh pts. in x and y
+  double s2u,s2v,s2e,s2p;
 
-  unorm_max = 0;
-  vnorm_max = 0;
-  pnorm_max = 0;
-  enorm_max = 0;
+#ifdef METAPHYSICL_HAVE_MASA
+  int err = 0;
+  double su,sv,sp,se;
+  double pnorm, unorm, vnorm, enorm;
+  double pnorm_max = 0., unorm_max = 0., vnorm_max = 0., enorm_max = 0.;
+  double prnorm_max = 0., urnorm_max = 0., vrnorm_max = 0., ernorm_max = 0.;
+#endif
 
   typedef VectorUnitVector<NDIM,0,RawScalar>::type XVector;
   XVector xvec = VectorUnitVector<NDIM,0,RawScalar>::value();

--- a/test/testopt_unit.C
+++ b/test/testopt_unit.C
@@ -30,7 +30,7 @@ int main(void)
   NumberArray<3, float> NA = 0;
   std::cos(NA);
 
-  NumberVector<3, float> NV = 0;
+  // NumberVector<3, float> NV = 0;
   // std::cos(NV);
 
   SparseNumberVectorUnitVector<3, 2, float>::type SNV;

--- a/test/testopt_unit.C
+++ b/test/testopt_unit.C
@@ -30,7 +30,9 @@ int main(void)
   NumberArray<3, float> NA = 0;
   std::cos(NA);
 
-  // NumberVector<3, float> NV = 0;
+  NumberVector<3, float> NV = 0;
+  // Avoid unused variable compiler warnings.
+  (void)NV;
   // std::cos(NV);
 
   SparseNumberVectorUnitVector<3, 2, float>::type SNV;

